### PR TITLE
refactor: dag-menu refactor

### LINKED ISSUE #10802: [Feature][UI] Task components are displayed in groups.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Task components are displayed in groups.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by labbomb**
## Purpose of the pull request
link [#10802](https://github.com/apache/dolphinscheduler/issues/10802)

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by songjianet — DISMISSED** [review]
LGTM

**Review by devosend — DISMISSED** [review]
LGTM

**Review by Amy0104 — DISMISSED** [review]
LGTM

**Review by devosend — DISMISSED** [review]
LGTM

**Review by songjianet — DISMISSED** [review]
LGTM

**Review by Amy0104 — DISMISSED** [review]
The front end part LGTM.

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql [hunk 1]
```diff
@@ -1953,11 +1953,11 @@ CREATE TABLE t_ds_cluster(
...
 
 --
--- Table structure for t_ds_fav
+-- Table structure for t_ds_fav_task
 --
 
-DROP TABLE IF EXISTS t_ds_fav;
-CREATE TABLE t_ds_fav
+DROP TABLE IF EXISTS t_ds_fav_task;
+CREATE TABLE t_ds_fav_task
```
Comment (zhongjiajie) [inline dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql:1960]: should we also change mysql DDL?

Comment (zhongjiajie) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql:1960]: oh I got it, in `dolphinscheduler_mysql.sql` we already with the name `t_ds_fav_task `

Comment (Tianqi-Dotes) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql:1960]: MySQL DDL already done before, h2 and postgre are forgot

**Review by zhongjiajie — APPROVED** [review]
LGTM, but we still need someone to take a look the front end part

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql (hunk 1 above)
Comment (zhongjiajie) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql:1960]: Thanks for your clarification

**Pull request merged**